src/transformers/models/blip/modeling_blip.py

```diff
@@ -1168,8 +1181,10 @@ def forward(
 
         >>> outputs = model(**inputs)
         ```"""
+        if labels is None and decoder_input_ids is None:
+            raise ValueError("Either `decoder_input_ids` or `labels` should be passed during inference.")
```

Comment:
It's weird that "labels" should be passed during inference?

Reply:
This is how it's done on the original impelemntation apprently, check: https://github.com/salesforce/BLIP/blob/3a29b7410476bf5f2ba0955827390eb6ea1f4f9d/models/blip_vqa.py#L46 --> `answer`

Reply:
Hmm I don't see it, the line you link to links to training mode

Reply:
Sorry what I meant is: https://github.com/salesforce/BLIP/blob/3a29b7410476bf5f2ba0955827390eb6ea1f4f9d/models/blip_vqa.py#L51-L69

Reply:
The code you link to is in the "training" mode right? So why would we have the warning that "labels should be passed during inference"? Do you mean training?

Reply:
Makes sense. I propose a clearer error message in [896bd63](https://github.com/huggingface/transformers/pull/21021/commits/896bd6334254495b447ebadcd453442579ebe2eb)